src/CoreWCF.WebHttp/src/CoreWCF/Configuration/ServiceModelWebServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using CoreWCF.Description;
+using CoreWCF.OpenApi;
+using Microsoft.Extensions.DependencyInjection;
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace CoreWCF.Configuration
+{
+    public static class ServiceModelWebServiceCollectionExtensions
+    {
+        public static IServiceCollection AddServiceModelWebServices(this IServiceCollection services, Action<OpenApiOptions> configure = null)
+        {
+            bool alreadyAdded = services.Any(s => s.ServiceType == typeof(IServiceBuilder));
+            if (!alreadyAdded)
+            {
+                services.AddServiceModelServices();
+            }
+                
+            services.AddSingleton<IServiceBehavior, WebHttpServiceBehavior>();
+            services.AddSingleton<ISwaggerProvider, OpenApiDocumentProvider>();
+
```

Comment:
In ServiceModelWebServiceBuilderExtensions you fetch ISwaggerProvider and then cast it to OpenApiDocumentProvider. The problem you might have is a different class also implementing ISwaggerProvider as well. All though I haven't checked their behavior, I would expect the Swashbuckle Swagger implementation to handle this by fetching `IEnumerable<ISwaggerProvider>`, and if they don't, I'd be suggesting it to them. So in ServiceModelWebServiceBuilderExtensions you will be wanting to get OpenApiDocumentProvider instead. You can handle this still being a singleton with this pattern:
```c#
services.AddSingleton<OpenApiDocumentProvider>();
services.AddSingleton<ISwaggerProvider>(provider => provider.GetRequiredService<OpenApiDocumentProvider>());
```
This way there's still only a singleton instance, but you can get it using either the interface or the concrete type and you avoid risking a cast exception in ServiceModelWebServiceBuilderExtensions.

Reply:
This has been addressed.